src/components/admin/reusables.tsx

```diff
@@ -0,0 +1,281 @@
+    }
+
+    setEditingMaterial(false);
+  };
+
+  /**
+   * Event handler delete icon click
+   *
+   * @param material material
+   */
+  const deleteIconClick = (material : ReusableMaterial) => () => {
+    setDeletableMaterial(material);
+    setDeletingMaterial(true);
+  };
+
+  /**
+   * Event handler edit icon click
+   *
+   * @param material material
+   */
+  const editIconClick = (material : ReusableMaterial) => () => {
+    setEditableMaterial(material);
+    setEditingMaterial(true);
+  };
+
+  /**
+   * Even handler for editable material name change
+   */
+  const handleEditableNameChange: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement> = ({ target }) => {
+    if (editableMaterial) {
+      setEditableMaterial({ ...editableMaterial, name: target.value });
+    }
+  };
+
+  /**
+   * Items for reusable material
+   * 
+   * @returns reusable material items
+   */
+  const reusableMaterialItems = () => (
+    materials.map(material =>
+      <MaterialItem key={ material.id }>
+        <MaterialText primary={ material.name }/>
+        <ListItemSecondaryAction>
+          <IconButton onClick={ deleteIconClick(material) }>
+            <Delete/>
+          </IconButton>
+          <IconButton onClick={ editIconClick(material) }>
+            <Edit/>
+          </IconButton>
+        </ListItemSecondaryAction>
+      </MaterialItem>
+    )
+  );
+
+  /**
+   * Renders add reusable dialog
+   */
+  const renderAddReusableMaterialDialog = () => (
+    <GenericDialog
+      error={ false }
+      open={ addingMaterial }
+      onClose={ () => setAddingMaterial(false) }
+      onCancel={ () => setAddingMaterial(false) }
+      onConfirm={ onAddReusableMaterialConfirm }
+      title={ strings.adminScreen.addNewReusableMaterialDialog.title }
+      positiveButtonText={ strings.generic.confirm }
+      cancelButtonText={ strings.generic.cancel }
+    >
+      <Typography variant="subtitle1">
+        { strings.adminScreen.addNewReusableMaterialDialog.text }
+      </Typography>
+      <TextField
+        color="secondary"
+        variant="standard"
+        onChange={ event => setNewMaterialName(event.target.value) }
+      />
+    </GenericDialog>
+  );
+
+  /**
+   * Renders delete reusable material dialog
+   */
+  const renderDeleteReusableMaterialDialog = () => (
+    <GenericDialog
+      error={ false }
+      open={ deletingMaterial }
+      onClose={ () => setDeletingMaterial(false) }
+      onCancel={ () => setDeletingMaterial(false) }
+      onConfirm={ onDeleteReusableMaterialConfirm }
+      title={ strings.adminScreen.deleteReusableMaterialDialog.title }
+      positiveButtonText={ strings.generic.confirm }
+      cancelButtonText={ strings.generic.cancel }
+    >
+      <Typography>
+        { strings.adminScreen.deleteReusableMaterialDialog.text }
+      </Typography>
+      <Typography variant="subtitle1">
+        { deletableMaterial?.name }
+      </Typography>
```

Comment:
These could be combined with strings.formatString() function.

Reply:
Fixed